crypto/keyring/keyring.go

```diff
@@ -719,7 +719,7 @@ func newRealPrompt(dir string, buf io.Reader) func(string) (string, error) {
 				continue
 			}
 
-			if err := os.WriteFile(dir+"/keyhash", passwordHash, 0o555); err != nil {
+			if err := os.WriteFile(dir+"/keyhash", passwordHash, 0o600); err != nil {
```

Comment:
Resolve this lint warning:

```
crypto/keyring/keyring.go:722:14: G306: Expect WriteFile permissions to be 0600 or less (gosec)
			if err := os.WriteFile(dir+"/keyhash", passwordHash, 0o555); err != nil {
```

Inspired by the implementation on Cosmos SDK main